### PR TITLE
Fix last snapshot index in delegated backup

### LINKED ISSUE #13932: Incorrect metric Master.EmbeddedJournalSnapshotLastIndex and Master.JournalEntriesSinceCheckPoint
**Alluxio Version:**
Master branch commit id  601272157bf

**Describe the bug**

The metric Master.EmbeddedJournalSnapshotLastIndex and Master.JournalEntriesSinceCheckPoint are not correct.

image

image

image

image

image

Log from alluxio primary master, so i guess the current sequence_number is about 5448746

```
2021-08-10 03:03:21,757 WARN  SnapshotReplicationManager - Failed to request snapshot info from followers: alluxio.exception.status.UnavailableException: No recent snapshot found from followers

2021-08-13 14:10:20,105 TRACE RaftJournalWriter - Writing entry 5448746: update_inode {
  id: 19
  last_modification_time_ms: 1628835020097
  last_access_time_ms: 1628835020097
}

2021-08-13 14:10:20,105 TRACE RaftJournalWriter - Flushing entry journal_entries {
  sequence_number: 5448746
  update_inode {
    id: 19
    last_modification_time_ms: 1628835020097
    last_access_time_ms: 1628835020097
  }
} (Message:ba021908aac8cc029a02...(size=28))
2021-08-13 14:10:20,107 TRACE RaftJournalWriter - Writing entry 5448747: inode_directory_id_generator {
  container_id: 36004
  sequence_number: 152943
}

2021-08-13 14:10:20,108 TRACE RaftJournalWriter - Flushing entry journal_entries {
  sequence_number: 5448747
  inode_directory_id_generator {
    container_id: 36004
    sequence_number: 152943
  }
}
 (Message:ba020f08abc8cc025208...(size=18))
```

**To Reproduce**

Not sure

**Expected behavior**

Correct metric, no warn at this time

**Urgency**

High

**Additional context**

No

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?
Fixes #13932
Fix the last snapshot related metrics 

### Why are the changes needed?
When delegation backup is implemented, the leader master will download the snapshot from standby master and use it as its own snapshot, but in this process, we didn't update the last snapshot related metrics of the leader master.

### Does this PR introduce any user facing changes?
Fix last snapshot metrics

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java [hunk 1]
```diff
@@ -216,6 +216,7 @@ public long takeSnapshot() {
         SAMPLING_LOG.warn("Failed to get raft group info: {}", e.getMessage());
       }
       long index = mSnapshotManager.maybeCopySnapshotFromFollower();
+      mSnapshotLastIndex = index;
```
Comment (maobaolong) [inline core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:219]: `index` maybe `-1`  while `DownloadState` is not one of `DOWNLOADED` `REQUEST_DATA` and `IDLE`

Comment (maobaolong) [reply core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:219]:
```suggestion
      if (index != RaftLog.INVALID_LOG_INDEX) {
        mSnapshotLastIndex = index;
      }
```

Comment (ggezer) [reply core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java:219]: Good after committing this change.

**Review by ggezer — APPROVED** [review]

**Comment by LuQQiu** [comment]
@maobaolong have you tested out the changes?
i guess the `long index = mSnapshotManager.maybeCopySnapshotFromFollower();` is either invalid index or a index bigger than the existing snapshot index?

**Comment by LuQQiu** [comment]
@yuzhu maybe we need to have some plans of how to add unit tests to cover the metrics failures? Metrics like this is so easy to be broken if we have new features go it

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**